app/views/issues/form/_form_actions.html.erb

```diff
@@ -0,0 +1,25 @@
+<div class="form-actions">
+  <div class="btn-group dropdown state-menu" data-behavior="state-menu">
+    <%= f.button :submit, nil, class: 'btn btn-primary text-capitalize', data: { behavior: 'state-submit-button' } %>
+    <%= f.hidden_field :state, data: { behavior: 'issue-state' } %>
+    <button type="button" class="btn btn-primary dropdown-toggle dropdown-toggle-split ml-0" data-toggle="dropdown">
+      <i class="fa fa-caret-down"></i>
+    </button>
+    <div class="dropdown-menu">
+      <div class="dropdown-item" data-state="draft">
+        <span class="state-header" data-state-header="draft"> Draft Issue</span>
+        <span class="state-details">If you feel there is still more work before this issue is ready.</span>
+      </div>
+      <div class="dropdown-item" data-state="review">
+        <span class="state-header" data-state-header="review"> Ready for Review Issue</span>
+        <span class="state-details">If you think this issue is ready and would like someone to have a look.</span>
+      </div>
+      <div class="dropdown-item" data-state="published">
+        <span class="state-header" data-state-header="published"> Published Issue</span>
+        <span class="state-details">If this is the final form of the issue and it’s ready for the report.</span>
+      </div>
+    </div>
+  </div>
+  or
+  <%= link_to 'Cancel', @issue.new_record? ? project_issues_path(current_project) : [current_project, @issue], class: 'cancel-link' %>
+</div>
```

Comment:
Since i changed the JS, i have to use the view logic below. For this to work, you'll have to add the below snippet to `Issue.rb`. We can either put the code here, or add it in a helper, or use a presenter. Whichever way works  too.

`app/models/issue.rb`
```ruby
  def self.state_descriptions
    {
      draft: 'Draft Issue',
      review: 'Ready for Review Issue',
      published: 'Published Issue'
    }
  end
```

`app/views/issues/form/_form_actions.html.erb`
We can even refactor the dropdown item logic further by including the long description in a helper and just loop over it.
```erb
<div class="form-actions">
  <div class="btn-group dropdown state-menu" data-behavior="state-menu">
    <%= f.button :submit, "#{@issue.new_record? ? 'Create' : 'Update'} #{Issue.state_descriptions[@issue.state.to_sym]}", class: 'btn btn-primary text-capitalize', data: { behavior: 'state-submit-button' } %>
    <%= f.hidden_field :state, data: { behavior: 'issue-state' } %>
    <button type="button" class="btn btn-primary dropdown-toggle dropdown-toggle-split ml-0" data-toggle="dropdown">
      <i class="fa fa-caret-down"></i>
    </button>
    <div class="dropdown-menu">
      <div class="dropdown-item <%= 'selected' if @issue.state == 'draft' %>" data-state="draft">
        <span class="state-header" data-state-header="draft"><%= @issue.new_record? ? 'Create' : 'Update' %> <%= Issue.state_descriptions[:draft] %></span>
        <span class="state-details">If you feel there is still more work before this issue is ready.</span>
      </div>
      <div class="dropdown-item <%= 'selected' if @issue.state == 'review' %>" data-state="review">
        <span class="state-header" data-state-header="review"><%= @issue.new_record? ? 'Create' : 'Update' %> <%= Issue.state_descriptions[:review] %></span>
        <span class="state-details">If you think this issue is ready and would like someone to have a look.</span>
      </div>
      <div class="dropdown-item <%= 'selected' if @issue.state == 'published' %>" data-state="published">
        <span class="state-header" data-state-header="published"><%= @issue.new_record? ? 'Create' : 'Update' %> <%= Issue.state_descriptions[:published] %></span>
        <span class="state-details">If this is the final form of the issue and it’s ready for the report.</span>
      </div>
    </div>
  </div>
  or
  <%= link_to 'Cancel', @issue.new_record? ? project_issues_path(current_project) : [current_project, @issue], class: 'cancel-link' %>
</div>
```

Reply:
Thanks @sean-yeoh. There already is a method for that: `Issue.state_names`. I've updated the JS and this file to use that method and simplify.